lib/compilers/ldc.js

```diff
@@ -49,6 +50,40 @@ class LDCCompiler extends BaseCompiler {
     isCfgCompiler() {
         return true;
     }
+
+    couldSupportASTDump(version) {
+        const versionRegex = /\((\d\.\d+)\.\d+\)/;
+        const versionMatch = versionRegex.exec(version);
+
+        if (versionMatch) {
+            const versionNum = parseFloat(versionMatch[1]);
+            return versionNum >= 1.4;
+        }
+
+        return false;
+    }
+
+    generateAST(inputFilename, options) {
+        // These options make LDC produce an AST dump in a separate file `<inputFilename>.cg`.
+        let newOptions = options.concat("-vcg-ast");
+        let execOptions = this.getDefaultExecOptions();
+        return this.runCompiler(this.compiler.exe, newOptions, this.filename(inputFilename), execOptions)
+            .then( () => { return this.loadASTOutput(this.filename(inputFilename)); });
+    }
+
+    loadASTOutput(inputFilename) {
+        // Load the AST output from the `.cg` file.
+        // Demangling is not needed.
+
+        let astPath = inputFilename.concat(".cg");
+        if (!fs.existsSync(astPath))
```

Comment:
Would there be a way to make this async?

Reply:
I had first tried to make the whole file loading operation async (I think), but it turns out the file is deleted quite soon, so using a ReaderStream etc. did not work. That's why I chose the simple `cat` approach used elsewhere too. This file exists check is not needed; better to have the actual `cat` operation error, I think. Can you tell me how to add error handling on the `bash -c cat` operation?

Reply:
To be honest, existSync is such a quick operation that I wouldn't bother.
The async method is actually depreciated inside node now so would actually need to use stat and that makes the code a bit more complex.

Reply:
Sounds good to me! Apologies for not replying sooner. I'm happy with this approach